### PR TITLE
Add public device keys to rageshakes

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Fixes https://github.com/element-hq/element-x-android/issues/2857

The `MatrixClientProvider` is now injected into `DefaultBugReporter`, to allow to access to new properties exposed in the `EncrytionService` (`deviceCurve25519()`, `deviceed25519()`)

I added a new test to assert the form data part added in the request by the BugReporter

 
## Type of change

- [ ] Feature
- [ ] Bugfix
- [x] Technical
- [ ] Other :

## Tests

- Step 1
- Step 2
- Step ...

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: features/rageshake/impl/src/main/kotlin/io/element/android/features/rageshake/impl/reporter/DefaultBugReporter.kt [hunk 1]
```diff
@@ -79,6 +82,7 @@ class DefaultBugReporter @Inject constructor(
     private val buildMeta: BuildMeta,
     private val bugReporterUrlProvider: BugReporterUrlProvider,
     private val sdkMetadata: SdkMetadata,
+    private val matrixClientsProvider: MatrixClientProvider,
```
Comment (bmarty) [inline features/rageshake/impl/src/main/kotlin/io/element/android/features/rageshake/impl/reporter/DefaultBugReporter.kt:85]:
```suggestion
    private val matrixClientProvider: MatrixClientProvider,
```

File: features/rageshake/impl/src/main/kotlin/io/element/android/features/rageshake/impl/reporter/DefaultBugReporter.kt [hunk 2]
```diff
@@ -156,6 +160,19 @@ class DefaultBugReporter @Inject constructor(
                         .addFormDataPart("user_id", userId)
                         .addFormDataPart("can_contact", canContact.toString())
                         .addFormDataPart("device_id", deviceId)
+                        .apply {
+                            userId.takeIf { MatrixPatterns.isUserId(it) }?.let {
+                                SessionId(it)
```
Comment (bmarty) [inline features/rageshake/impl/src/main/kotlin/io/element/android/features/rageshake/impl/reporter/DefaultBugReporter.kt:165]: `sessionData?.userId` is guaranteed to be a valid userId, so to simplify, I would change line 148 from 
`val userId = sessionData?.userId ?: "undefined"`
to
`val userId = sessionData?.userId?.let { UserId(it) }` 

and use `?: "undefined"` at line 148.

Please be aware that the value can still be null (user can send bug report before having a session.

**Review by bmarty — COMMENTED** [review]
Thanks for the new test!
I think EXPECTED_NUMBER_OF_PROGRESS_VALUE will have to be changed to 18

File: features/rageshake/impl/src/main/kotlin/io/element/android/features/rageshake/impl/reporter/DefaultBugReporter.kt (hunk 1 above)
Comment (BillCarsonFr) [reply features/rageshake/impl/src/main/kotlin/io/element/android/features/rageshake/impl/reporter/DefaultBugReporter.kt]: done

File: features/rageshake/impl/src/main/kotlin/io/element/android/features/rageshake/impl/reporter/DefaultBugReporter.kt (hunk 2 above)
Comment (BillCarsonFr) [reply features/rageshake/impl/src/main/kotlin/io/element/android/features/rageshake/impl/reporter/DefaultBugReporter.kt]: Updated

**Comment by BillCarsonFr** [comment]
> Thanks for the new test! I think EXPECTED_NUMBER_OF_PROGRESS_VALUE will have to be changed to 18

The existing test was happy because we don't add that part if the keys are null. So I added the listener in the other test to ensure that we have one more thing reported

**Review by bmarty — APPROVED** [review]
Thanks for the update!

**Pull request merged**